server/src/main/java/org/candlepin/pinsetter/core/PinsetterKernel.java

```diff
@@ -551,6 +562,37 @@ private void deleteAllJobs() {
         return scheduler.getJobKeys(GroupMatcher.jobGroupEquals(SINGLE_JOB_GROUP));
     }
 
+    public void retriggerCronJob(String taskName, Class<? extends KingpinJob> jobClass)  throws
+        PinsetterException {
+        Set<TriggerKey> cronTriggerKeys = null;
+        try {
+            cronTriggerKeys = scheduler.getTriggerKeys(
+                GroupMatcher.triggerGroupEquals(PinsetterKernel.CRON_GROUP));
+            TriggerKey key = null;
+            Iterator<TriggerKey> keysTrigger = cronTriggerKeys.iterator();
+            while (key == null && keysTrigger.hasNext()) {
+                TriggerKey current = keysTrigger.next();
+                if (current.getName().contains(taskName)) {
+                    key = current;
```

Comment:
If I'm reading this correctly, we're just looping through all the cronTriggerKeys looking for one that matches `taskName`.  Are we expecting more than one Key to match?  If not, we can add a `break`.  If so, is the order of the Keys deterministic?

Reply:
@awood  we do not expect more than one key to match.
Also, this is a while loop which would stop immediately after _key_ is assigned. I dont like to use _break_ so I went this way for the same purpose. should I add a comment to specify this information?